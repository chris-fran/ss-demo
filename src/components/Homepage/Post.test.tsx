import initStoryshots from '@storybook/addon-storyshots';
import { imageSnapshot } from '@storybook/addon-storyshots-puppeteer';
import { shallow } from 'enzyme';

initStoryshots({
  suite: 'Serialized snapshots',
  storyKindRegex: /^Post$/,
  renderer: shallow,
});

initStoryshots({
  suite: 'Image snapshots',
  storyKindRegex: /^Post$/,
  test: imageSnapshot(),
});
