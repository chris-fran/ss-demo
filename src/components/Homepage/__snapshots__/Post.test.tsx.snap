// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Post created by me 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Box>
    <Post
      body="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor."
      createdAt={2019-02-24T00:21:15.925Z}
      guid="45b8b7af-75d2-4711-904c-5802b1c8e342"
      isMine={true}
      user={
        Object {
          "avatarUrl": "https://via.placeholder.com/96x96",
          "name": "Christopher Francisco",
          "username": "ReijiS2",
        }
      }
    />
  </Box>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Post
        body="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor."
        createdAt={2019-02-24T00:21:15.925Z}
        guid="45b8b7af-75d2-4711-904c-5802b1c8e342"
        isMine={true}
        user={
          Object {
            "avatarUrl": "https://via.placeholder.com/96x96",
            "name": "Christopher Francisco",
            "username": "ReijiS2",
          }
        }
      />,
      "style": Object {
        "width": 600,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "body": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor.",
        "createdAt": 2019-02-24T00:21:15.925Z,
        "guid": "45b8b7af-75d2-4711-904c-5802b1c8e342",
        "imageUrl": undefined,
        "isMine": true,
        "user": Object {
          "avatarUrl": "https://via.placeholder.com/96x96",
          "name": "Christopher Francisco",
          "username": "ReijiS2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Post
          body="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor."
          createdAt={2019-02-24T00:21:15.925Z}
          guid="45b8b7af-75d2-4711-904c-5802b1c8e342"
          isMine={true}
          user={
            Object {
              "avatarUrl": "https://via.placeholder.com/96x96",
              "name": "Christopher Francisco",
              "username": "ReijiS2",
            }
          }
        />,
        "style": Object {
          "width": 600,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "body": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor.",
          "createdAt": 2019-02-24T00:21:15.925Z,
          "guid": "45b8b7af-75d2-4711-904c-5802b1c8e342",
          "imageUrl": undefined,
          "isMine": true,
          "user": Object {
            "avatarUrl": "https://via.placeholder.com/96x96",
            "name": "Christopher Francisco",
            "username": "ReijiS2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Storyshots Post default 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Box>
    <Post
      body="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor."
      createdAt={2019-02-24T00:21:15.925Z}
      guid="45b8b7af-75d2-4711-904c-5802b1c8e342"
      isMine={false}
      user={
        Object {
          "avatarUrl": "https://via.placeholder.com/96x96",
          "name": "Christopher Francisco",
          "username": "ReijiS2",
        }
      }
    />
  </Box>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Post
        body="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor."
        createdAt={2019-02-24T00:21:15.925Z}
        guid="45b8b7af-75d2-4711-904c-5802b1c8e342"
        isMine={false}
        user={
          Object {
            "avatarUrl": "https://via.placeholder.com/96x96",
            "name": "Christopher Francisco",
            "username": "ReijiS2",
          }
        }
      />,
      "style": Object {
        "width": 600,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "body": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor.",
        "createdAt": 2019-02-24T00:21:15.925Z,
        "guid": "45b8b7af-75d2-4711-904c-5802b1c8e342",
        "imageUrl": undefined,
        "isMine": false,
        "user": Object {
          "avatarUrl": "https://via.placeholder.com/96x96",
          "name": "Christopher Francisco",
          "username": "ReijiS2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Post
          body="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor."
          createdAt={2019-02-24T00:21:15.925Z}
          guid="45b8b7af-75d2-4711-904c-5802b1c8e342"
          isMine={false}
          user={
            Object {
              "avatarUrl": "https://via.placeholder.com/96x96",
              "name": "Christopher Francisco",
              "username": "ReijiS2",
            }
          }
        />,
        "style": Object {
          "width": 600,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "body": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor.",
          "createdAt": 2019-02-24T00:21:15.925Z,
          "guid": "45b8b7af-75d2-4711-904c-5802b1c8e342",
          "imageUrl": undefined,
          "isMine": false,
          "user": Object {
            "avatarUrl": "https://via.placeholder.com/96x96",
            "name": "Christopher Francisco",
            "username": "ReijiS2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Storyshots Post with image 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Box>
    <Post
      body="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor."
      createdAt={2019-02-24T00:21:15.925Z}
      guid="45b8b7af-75d2-4711-904c-5802b1c8e342"
      imageUrl="https://via.placeholder.com/600x450"
      isMine={false}
      user={
        Object {
          "avatarUrl": "https://via.placeholder.com/96x96",
          "name": "Christopher Francisco",
          "username": "ReijiS2",
        }
      }
    />
  </Box>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Post
        body="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor."
        createdAt={2019-02-24T00:21:15.925Z}
        guid="45b8b7af-75d2-4711-904c-5802b1c8e342"
        imageUrl="https://via.placeholder.com/600x450"
        isMine={false}
        user={
          Object {
            "avatarUrl": "https://via.placeholder.com/96x96",
            "name": "Christopher Francisco",
            "username": "ReijiS2",
          }
        }
      />,
      "style": Object {
        "width": 600,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "body": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor.",
        "createdAt": 2019-02-24T00:21:15.925Z,
        "guid": "45b8b7af-75d2-4711-904c-5802b1c8e342",
        "imageUrl": "https://via.placeholder.com/600x450",
        "isMine": false,
        "user": Object {
          "avatarUrl": "https://via.placeholder.com/96x96",
          "name": "Christopher Francisco",
          "username": "ReijiS2",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Post
          body="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor."
          createdAt={2019-02-24T00:21:15.925Z}
          guid="45b8b7af-75d2-4711-904c-5802b1c8e342"
          imageUrl="https://via.placeholder.com/600x450"
          isMine={false}
          user={
            Object {
              "avatarUrl": "https://via.placeholder.com/96x96",
              "name": "Christopher Francisco",
              "username": "ReijiS2",
            }
          }
        />,
        "style": Object {
          "width": 600,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "body": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam auctor nunc non metus eleifend, id ornare nulla sollicitudin. Phasellus blandit lacus vitae ullamcorper lobortis. Vivamus in neque tortor.",
          "createdAt": 2019-02-24T00:21:15.925Z,
          "guid": "45b8b7af-75d2-4711-904c-5802b1c8e342",
          "imageUrl": "https://via.placeholder.com/600x450",
          "isMine": false,
          "user": Object {
            "avatarUrl": "https://via.placeholder.com/96x96",
            "name": "Christopher Francisco",
            "username": "ReijiS2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
